### PR TITLE
fix(bulma-ui): fix standalone Badge pointer-events, pulse halo, and falsy content

### LINKED ISSUE #264: [Bug] Standalone Badge is inert (pointer-events), pulse halo escapes, falsy content renders an empty pill
## Summary

Deep review of merged #257 found four related Badge defects, all in **standalone mode** (no
`children`) and all invisible to jsdom (fireEvent bypasses hit-testing; jsdom doesn't lay out
CSS), which is why four review passes missed them.

### 1. Standalone badge swallows all pointer events (major)

`bulma-ui/src/scss/components/_badge.scss:55` sets `pointer-events: none` on the base `.badge`
(correct for the anchored overlay so clicks pass through to the child), but the
`.is-standalone` reset (`_badge.scss:96`) restores only `position/display/transform` — never
`pointer-events`. In standalone mode the pill **is** the component root and `Badge.tsx`
deliberately spreads `rest` onto it so callers can attach handlers.

**Repro:** `<Badge content={5} onClick={dismiss} title="unread" />` — the click handler never
fires and the tooltip never shows in a real browser; the browser retargets every pointer event
to whatever is underneath.

**Fix:** add `pointer-events: auto;` to the `.is-standalone` reset. Regression test note: a
jsdom `fireEvent.click` test passes even while this is broken — assert the computed style
(`pointer-events`) instead, or cover via a Storybook/browser check.

### 2. Standalone + `pulse`: the halo escapes and covers the nearest positioned ancestor (major)

`.is-standalone` sets `position: static` (`_badge.scss:97`), but `.is-pulse::after` is
`position: absolute; inset: 0` (`_badge.scss:157-159`). With the badge static, the pseudo
element resolves against the nearest positioned ancestor.

**Repro:** `<Badge content={5} pulse />` inside any `position: relative` container (Card,
Navbar item, …) paints a badge-colored overlay animating opacity/scale across the **entire
container** every 1.4s instead of a small ring around the pill.

**Fix:** use `position: relative` (not `static`) in the `.is-standalone` reset so the `::after`
anchors to the pill. (Distinct from the transform issue fixed in #257's `3e2e44c`.)

### 3. Falsy-but-non-null `content` renders a visible empty pill

`Badge.tsx` (~line 102): `hasContent = content != null && (!isZero || showZero)` only excludes
`null`/`undefined` (and `0` without `showZero`). The common idiom
`<Badge content={count > 0 && count}>…</Badge>` yields `content={false}` at zero: `isZero` is
false (typeof isn't `'number'`), `hasContent` is true, and the user sees a small empty colored
blob with `role="status"` and no label. `content=''` and `content={true}` behave the same.
The same file already guards the analogous children case (`children !== false`).

**Fix:** exclude `false`, `true`, and `''` from `hasContent`; add tests for all three.

### 4. `max` is unsanitized while sibling `Avatars` sanitizes (minor)

`Badge.tsx` (~line 109): `content > max ? \`${max}+\` : String(content)` — so
`<Badge content={0} max={-1} showZero />` renders **"-1+"** (and `aria-label="-1+"`).
`Avatars.tsx` in the same PR sanitizes its `max` (`Number.isInteger(max) && max >= 0`).

**Fix:** apply the same sanitization; add the `content === max` and negative-`max` boundary
tests.

### 5. `.badge-wrapper` `line-height: 0` collapses plain-text children (minor)

`_badge.scss:20`: the wrapper is `inline-flex` with `line-height: 0`; a plain text child
(`<Badge content={3}>Inbox</Badge>`) computes to height 0 (verified by headless-Chrome
measurement: 0 vs 18px without the rule).

**Fix:** scope the line-height reset so it doesn't zero out text children (e.g. apply it to
the indicator, not the wrapper), with a story/docs example using a text child.

## Definition of done

- SCSS fixes follow the register-vars pattern (`bulma-ui/src/scss/CLAUDE.md`); no literals.
- Tests keep bulma-ui at ≥99% all metrics; boundary cases above covered.
- Stories updated where behavior is visible (standalone interactive badge, standalone pulse).
- No docs/API surface changes expected; run `pnpm gen:catalog` regardless.

Found by adversarially-verified multi-agent review of the merged #257 diff.

### PR DESCRIPTION
**Pull request by bestaxbot**
## Summary

Deep review of merged #257 found four related Badge defects, all in standalone mode (no
`children`) and all invisible to jsdom (`fireEvent` bypasses hit-testing; jsdom doesn't lay
out CSS or resolve inheritance), which is why they slipped past review.

- **Pointer events**: `.badge` resets `pointer-events: none` (correct for the anchored overlay
  so clicks pass through to the child), but `.is-standalone` never restored it, so a standalone
  badge with an `onClick` never receives clicks in a real browser. Fixed by adding
  `pointer-events: auto` to `.is-standalone`.
- **Pulse halo escape**: `.is-standalone` set `position: static`, but `.is-pulse::after` is
  `position: absolute; inset: 0` — with the badge static, the halo resolves against the
  nearest positioned ancestor (e.g. the whole Card/Navbar it sits in) instead of the pill.
  Fixed by using `position: relative` instead of `static`.
- **Falsy content renders an empty pill**: `hasContent` only excluded `null`/`undefined` (and
  unshown `0`), so the common `content={count > 0 && count}` idiom renders a visible, empty,
  `role="status"` pill at `content={false}`. `content={true}` and `content=""` behaved the
  same. Fixed by excluding `false`/`true`/`''` from `hasContent`.
- **Unsanitized `max`**: `content > max` used `max` directly, so e.g. `max={-1}` rendered
  `"-1+"` instead of falling back sensibly. Fixed by sanitizing `max` the same way `Avatars`
  already sanitizes its `max` (falls back to the default `99` when not a non-negative integer).
- **`.badge-wrapper` `line-height: 0`**: inherited into plain-text `children` (e.g.
  `<Badge content={3}>Inbox</Badge>`) and collapsed their line box. The badge pill already
  declares its own `line-height: 1`, so the wrapper-level reset wasn't needed — removed it.

## Tests

- Bugs 3 & 4 (JS logic) get standard failing→passing Jest coverage in `Badge.test.tsx`.
- Bugs 1 & 2 (CSS-only, invisible to jsdom's DOM/className assertions) get a new
  `Badge.styles.test.tsx` that compiles the real `_badge.scss` partial with `sass` (already a
  devDependency) and injects it into jsdom, then asserts `getComputedStyle` — this reproduces
  both bugs (verified failing against the pre-fix code, now passing).
- Bug 5 (`line-height` inheritance) isn't reproducible in jsdom, which doesn't implement CSS
  inheritance resolution for `getComputedStyle` on elements without a matching rule (verified:
  a text child's computed `line-height` comes back as an empty string either way). Covered
  instead by a new Storybook story (`TextChild`) and docs example, per the issue's own guidance
  ("or cover via a Storybook/browser check").
- Added `StandaloneInteractive` and `StandalonePulse` stories/docs examples exercising the two
  browser-only fixes.

Coverage stays at 100% for `Badge.tsx`.

Fixes #264

Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

- **New Features**
  - Added new interactive standalone badge stories, including click behavior, pulsing badges, and badges with plain text labels.

- **Bug Fixes**
  - Improved badge rendering rules so additional “empty” values no longer display a badge.
  - Sanitized `max` for invalid values to reliably fall back to the default, including correct `{max}+`/exact-number rendering.
  - Updated standalone/pulse styling to ensure correct pointer interaction and prevent pulse decoration from blocking clicks.

- **Documentation**
  - Clarified `max` prop behavior and expanded standalone usage guidance.

### CODE REVIEW

File: bulma-ui/src/scss/components/_badge.scss [hunk 1]
```diff
@@ -93,10 +92,17 @@ $badge-animation-duration: 1.4s !default;
...
+//   whatever positioned container the badge sits in.
+// - pointer-events: auto overrides the base .badge reset (needed for the
+//   anchored overlay so clicks pass through to the child) because in
+//   standalone mode the pill itself is the interactive root.
 .#{iv.$class-prefix}badge.#{iv.$class-prefix}is-standalone {
-  position: static;
+  position: relative;
   display: inline-flex;
   transform: none;
+  pointer-events: auto;
```
[the remark this answers is not in the document]
Comment (bestaxbot) [reply bulma-ui/src/scss/components/_badge.scss:105]: Fixed in f3ecb613a1339dfbffde3f9ea41c6f0fbf3ecd7d — you're right, this is a real regression introduced by this PR. `pointer-events` is inherited, so once `.is-standalone` flipped the pill to `auto` (line 105) the `.is-pulse::after` halo (which declares no `pointer-events` of its own) inherited `auto` too. Since the halo is `position: absolute; inset: 0` and animates to `scale(1.75)` while fading to `opacity: 0` (which doesn't disable hit-testing), a standalone pulse badge captured clicks across a box ~37.5% larger than the pill on every edge — exactly the invisible click-blocker you describe, and the new `StandalonePulse` story would exhibit it. In the anchored case the base `.badge` keeps `pointer-events: none` so the halo was already inert there; only standalone regressed.

I added `pointer-events: none;` to `.badge.is-pulse::after` in `_badge.scss` so the decorative halo never hit-tests while the pill stays interactive. I also added a test in `Badge.styles.test.tsx` that compiles the real partial and asserts the pulse rule declares `pointer-events: none` — I went that route because jsdom's `getComputedStyle` ignores the pseudo-element argument and returns the base element's style, so it can't observe the `::after` cascade directly.

File: bulma-ui/src/scss/components/_badge.scss [hunk 2]
```diff
@@ -158,6 +164,10 @@ $badge-animation-duration: 1.4s !default;
   content: '';
   position: absolute;
   inset: 0;
+  // Decorative halo only — never intercept clicks. pointer-events is inherited,
```
[the remark this answers is not in the document]
Comment (bestaxbot) [reply bulma-ui/src/scss/components/_badge.scss:168]: Good catch — added the blank line before the comment in 526190a. The repo doesn't actually run stylelint in CI (there's no stylelint config here, so `scss/double-slash-comment-empty-line-before` isn't enforced on our side), but the blank line is a clearly-correct, harmless improvement and it matches this file's own convention — every other `//` comment in `_badge.scss` (e.g. the "Pulse animation" comment just above on line 162) is preceded by a blank line, so the mid-block comment was the odd one out.
